stac_model/examples.py

```diff
@@ -0,0 +1,156 @@
+            333.00778264,
+            395.09249139,
+            593.75055589,
+            566.4170017,
+            861.18399006,
+            1086.63139075,
+            1117.98170791,
+            404.91978886,
+            4.77584468,
+            1002.58768311,
+            761.30323499,
+            1231.58581042,
+        ],
+    )
+    mlm_input = ModelInput(
+        name="13 Band Sentinel-2 Batch",
+        bands=band_names,
+        input_array=input_array,
+        norm_by_channel=True,
+        norm_type="z_score",
+        resize_type="none",
+        statistics=stats,
+        pre_processing_function="https://github.com/microsoft/torchgeo/blob/545abe8326efc2848feae69d0212a15faba3eb00/torchgeo/datamodules/eurosat.py",  # noqa: E501
+    )
+    mlm_runtime = Runtime(
+        framework="torch",
+        version="2.1.2+cu121",
+        asset=Asset(
+            href="https://huggingface.co/torchgeo/resnet18_sentinel2_all_moco/resolve/main/resnet18_sentinel2_all_moco-59bfdff9.pth"  # noqa: E501
+        ),
+        source_code=Asset(
+            href="https://github.com/microsoft/torchgeo/blob/61efd2e2c4df7ebe3bd03002ebbaeaa3cfe9885a/torchgeo/models/resnet.py#L207"  # noqa: E501
+        ),
+        accelerator="cuda",
+        accelerator_constrained=False,
+        hardware_summary="Unknown",
+    )
+    result_array = ResultArray(
+        shape=[-1, 10], dim_names=["batch", "class"], data_type="float32"
+    )
+    class_map = {
+        "Annual Crop": 0,
+        "Forest": 1,
+        "Herbaceous Vegetation": 2,
+        "Highway": 3,
+        "Industrial Buildings": 4,
+        "Pasture": 5,
+        "Permanent Crop": 6,
+        "Residential Buildings": 7,
+        "River": 8,
+        "SeaLake": 9,
+    }
+    class_objects = [
+        ClassObject(value=class_map[class_name], name=class_name)
+        for class_name in class_map
+    ]
+    mlm_output = ModelOutput(
+        task="classification",
+        classification_classes=class_objects,
+        output_shape=[-1, 10],
+        result_array=[result_array],
+    )
+    ml_model_meta = MLModelProperties(
+        mlm_name="Resnet-18 Sentinel-2 ALL MOCO",
+        mlm_task="classification",
+        mlm_framework="pytorch",
+        mlm_framework_version="2.1.2+cu121",
+        mlm_file_size=43000000,
+        mlm_memory_size=1,
+        mlm_summary=(
+            "Sourced from torchgeo python library,"
+            "identifier is ResNet18_Weights.SENTINEL2_ALL_MOCO"
+        ),
+        mlm_pretrained_source="EuroSat Sentinel-2",
+        mlm_total_parameters=11_700_000,
+        mlm_input=[mlm_input],
+        mlm_runtime=[mlm_runtime],
+        mlm_output=[mlm_output],
+    )
+    # TODO, this can't be serialized but pystac.item calls for a datetime
+    # in docs. start_datetime=datetime.strptime("1900-01-01", "%Y-%m-%d")
+    # Is this a problem that we don't do date validation if we supply as str?
+    start_datetime = "1900-01-01"
+    end_datetime = None
```

Comment:
will it return an object that can be serialized, unlike datetime.strptime? it looks like it also returns a datetime object: https://dateutil.readthedocs.io/en/stable/parser.html#dateutil.parser.parse

Reply:
It will return a `datetime.datetime` object. I assume pydantic is able to handle it? Otherwise, a custom serializer calling `datetime.isoformat()` explicitly might be needed.

Reply:
Looks like pydantic should natively support it: 
https://docs.pydantic.dev/latest/concepts/serialization/#modelmodel_dump_json

Reply:
Indeed, `pydantic` will not do validation, but `pystac.validator` will test it against the STAC Item core schema.
You might need to call it explicitly though. 

I wonder if it would be possible to pass `pystac.validator`  as a hook to `pydantic` to automate it?

Reply:
I think we need to bind custom validators to each pystac extension we want to bring into a nested MLModel object? https://docs.pydantic.dev/latest/concepts/validators/ these validators would bind to the types and could then be used anywhere I think?

Reply:
```
Plain validators are like a mode='before' validator but they terminate validation immediately, no further validators are called and Pydantic does not do any of its internal validation.
```
from https://docs.pydantic.dev/latest/concepts/validators/#before-after-wrap-and-plain-validators

maybe we patch in the pystac validation for each extension as a plain validator

Reply:
I'll give this a try with non mlm fields